azurelinuxagent/common/protocol/wire.py

```diff
@@ -797,20 +797,26 @@ def update_goal_state(self, forced=False):
         except Exception as exception:
             raise ProtocolError("Error fetching goal state: {0}".format(ustr(exception)))
 
-    def update_extensions_goal_state(self):
+    def update_extensions_goal_state(self, wait_for_new_goal_state=True):
```

Comment:
This new parameter is there just for the benefit of the unit tests. In a couple of iterations of this code the retry loop and this parameter should go away. Right now the loop is being used to debug an issue in the hostgaplugin.